pkg/api/types.go

```diff
@@ -2785,6 +2802,12 @@ type DeleteOptions struct {
 	// finalizer will be added to/removed from the object's finalizers list.
 	// +optional
 	OrphanDependents *bool
+
+	// Whether and how garbage collection will be performed.
+	// Defaults to DeletePropagationDefault.
+	// Either this field or OrphanDependents may be set, but not both.
+	// +optional
+	PropagationPolicy *DeletePropagationPolicy
```

Comment:
How do you pass a non-basic type to a rest api call? I think what is done for other types like this is a function that expand the non-basic types to its elements and pass each element as a basic query parameter to the api call.